os/fs/driver/mtd/smart.c

```diff
@@ -5109,13 +5147,169 @@ static int smart_joutnal_set_area_state(FAR struct smart_struct_s *dev, int area
 	ret = MTD_WRITE(dev->mtd, readaddress, 1, &id_bits);
 	if (ret != 1) {
 		fdbg("Error updating physical sector %d status\n", psector);
-		return -EIO;
+		return ret;
 	}
 
 	dev->journal_area_state[area] = id_bits;
 	return OK;
 }
 
+/****************************************************************************
+ * Name: smart_journal_get_writeaddress
+ *
+ * Description:
+ *   Get address to write next journal data
+ *
+ ****************************************************************************/
+
+static int smart_journal_get_writeaddress(FAR struct smart_struct_s *dev, uint32_t *address)
+{
+	uint16_t psector;
+	size_t seq;
+	uint32_t *offset = address;
+
+	psector = dev->neraseblocks * dev->sectorsPerBlk;
+
+	/* Get Physical sector & sequence of each area First */
+	if (dev->area == 0) {
+		if (dev->journal_seq >= dev->njournaldata / 2) {
```

Comment:
It might be better to use dev->njournaldata >> 1 instead of divide operation.

Reply:
Good point, I will apply it everywhere, thanks.